lib/galaxy/webapps/galaxy/services/invocations.py

```diff
@@ -135,6 +138,54 @@ def show(self, trans, invocation_id, serialization_params, eager=False):
         wfi = self._workflows_manager.get_invocation(trans, invocation_id, eager)
         return self.serialize_workflow_invocation(wfi, serialization_params)
 
+    def get_metrics(
+        self,
+        trans,
+        invocation_id: DecodedDatabaseIdField,
+    ) -> MetricsSummaryCumulative:
+        workflow_invocation = self._workflows_manager.get_invocation(trans, invocation_id, eager=True)
+        job_ids = [step.job_id for step in workflow_invocation.steps if step.job_id is not None]
```

Comment:
That's excluding all subworkflows jobs and all map over jobs.

Reply:
I've got a rudimentary solution to accessing subworkflow jobs. Am I on the right track here, or is there a recommended way to access subworkflow jobs?

```python
job_ids = [step.job_id for step in workflow_invocation.steps if step.job_id is not None]

for subworkflow in workflow_invocation.subworkflow_invocations:
    job_ids.extend([step.job_id for step in subworkflow.steps if step.job_id is not None])
```

I'm actually unfamiliar with what map over jobs are. Could you please clarify what they are? :)